src/iam.ts

```diff
@@ -272,6 +277,27 @@ export class IAM extends IAMBase {
     throw new Error(ERROR_MESSAGES.DID_DOCUMENT_NOT_INITIALIZED);
   }
 
+  async updateAssetDidDocument(options: {
+    did: string,
+    value: IAttributePayload
```

Comment:
Yes, I think it would be better if this was more generic and it accepted an `IUpdateData` param.